### PR TITLE
[ AM-27 ] Adding 'authProvider' field for operator for third party authentication

### PR DESCRIPTION
**Pull request by stephendeyoung**

### CODE REVIEW

File: thng-resource-model/src/main/java/com/evrythng/thng/resource/model/store/Operator.java [hunk 1]
```diff
@@ -10,6 +10,10 @@
  */
 public class Operator extends AbstractUser {
 
+	public enum AuthProvider {
+		google
```
Comment (dorogush) [inline thng-resource-model/src/main/java/com/evrythng/thng/resource/model/store/Operator.java]: Must be all uppercase. 
https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names

Comment (stephendeyoung) [reply thng-resource-model/src/main/java/com/evrythng/thng/resource/model/store/Operator.java]: If I do this when creating an operator a client would have to specify:

```
{
  "authProvider": "GOOGLE"
}
```

Is there some way I can avoid the client needing to do this?

Comment (dorogush) [reply thng-resource-model/src/main/java/com/evrythng/thng/resource/model/store/Operator.java]: https://github.com/evrythng/evrythng-java-sdk/blob/master/thng-resource-model/src/main/java/com/evrythng/thng/resource/model/store/AbstractUser.java#L216

Comment (stephendeyoung) [reply thng-resource-model/src/main/java/com/evrythng/thng/resource/model/store/Operator.java]: Thanks. I've made the changes.

File: thng-resource-model/src/main/java/com/evrythng/thng/resource/model/store/Operator.java [hunk 2]
```diff
@@ -5,11 +5,46 @@
...
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import java.util.Map;
+
 /**
  * Model representation for <em>operator</em>.
  */
 public class Operator extends AbstractUser {
 
+	public enum AuthProvider {
```
Comment (YuryYaroshevich) [inline thng-resource-model/src/main/java/com/evrythng/thng/resource/model/store/Operator.java]: I would extract that enum into separate class

**Comment by stephendeyoung** [comment]
@YuryYaroshevich the formatting is in line with the rest of the file as this project has not been used with the Google Java formatting before

File: thng-resource-model/src/main/java/com/evrythng/thng/resource/model/store/Operator.java (hunk 2 above)
Comment (stephendeyoung) [reply thng-resource-model/src/main/java/com/evrythng/thng/resource/model/store/Operator.java]: Other classes in this project already have nested classes: https://github.com/evrythng/evrythng-java-sdk/blob/master/thng-resource-model/src/main/java/com/evrythng/thng/resource/model/store/AbstractUser.java#L155

But I can put this in its own class if you still think that's best.

Comment (YuryYaroshevich) [reply thng-resource-model/src/main/java/com/evrythng/thng/resource/model/store/Operator.java]: I would do this but it is not critical for me. No problem if you will leave it as it is right now.

Comment (stephendeyoung) [reply thng-resource-model/src/main/java/com/evrythng/thng/resource/model/store/Operator.java]: I've extracted the enum into a separate class Yury.

**Pull request merged**